packages/components/src/internal/components/editable/actions.ts

```diff
@@ -654,9 +654,9 @@ export function removeColumn(
 async function prepareUpdateRowDataFromBulkForm(
     queryInfo: QueryInfo,
     rowData: OrderedMap<string, any>,
-    isIncludedColumn: (col: QueryColumn) => boolean,
-    containerPath: string,
-    altColumns?: string[]
+    isIncludedColumn?: (col: QueryColumn) => boolean,
+    containerPath?: string,
+    altColumns?: string[] // TODO: This should use the same metadata for columns as the rest of the editable grid
```

Comment:
Using `?` is just as much about telling users its optional as it is telling composers/editors of the function that it is optional and they need account for it not being defined. Of course any parameter could not be defined but in practice we expect parameters that are not optional to be defined.

> Yes, upstream they may be passed as undefined, but they're always passed. IMO we should only be marking arguments as optional if we expect consumers to use them without passing the args

Whether or not they caller explicitly passed the optional arguments is superfluous to the implementor. It is understood within the function that those arguments are optional. In the case we're looking at here some/all of these parameters are passed through. Why would the above method require the arguments when the below method does not require them?

From a callers perspective, I much prefer being as concise in calls as possible. `doThing(1, 0)` instead of `doThing(1, 0, undefined)` since `doThing(first: number, second: number, third?: number)` already let me know I don't need to pass arguments.

> If we were to stop making things optional we could simplify a lot of our code, by not having to check if something is defined or not.

I don't see how in this case it would simplify these checks and in turn seems to only weaken the interface contract making our code have to be likely even more defensive (e.g. logically I'd have to assume every parameter is not defined so effectively assume every parameter is marked as `?`).

Reply:
> Optional arguments are not the same as `@Nullable`, they're more equivalent to adding an alternate java method with a different signature that defaults the `@Nullable` args as `null`.

I'd argue it is both like adding different signatures AND like `@Nullable` but replace `null` with `undefined`.